src/services/activityReports.js

```diff
@@ -228,7 +228,92 @@ export function activityReports() {
       },
       {
         model: User,
-        attributes: ['name', 'role', 'fullName'],
+        attributes: ['id', 'name', 'role', 'fullName'],
+        as: 'collaborators',
+        through: { attributes: [] },
+      },
+    ],
+  });
+}
+/**
+ * Retrieves alerts based on the following logic:
+ * One or both of these high level conditions are true -
+ * manager - approvingManagerId matches and report status is submitted.
+ * specialist - author id or one of the collaborator's id matches and status is not approved nor
+ * submitted.
+ * @param {*} userId
+ */
+export function activityReportAlerts(userId) {
+  return ActivityReport.findAll({
+    where: {
+      [Op.or]: [
+        {
+          [Op.or]: [{ status: 'submitted' }, { status: 'needs_action' }],
+          approvingManagerId: userId,
+        },
+        {
+          [Op.and]: [
+            {
+              [Op.and]: [
+                {
+                  status: { [Op.ne]: 'approved' },
```

Comment:
Do any of these status strings exist in a Constant? It'd be good to refer to the constants rather than hard coding in the query.

Reply:
Good catch!
Fixed.